nasl/nasl_scanner_glue.c

```diff
@@ -1077,6 +1078,36 @@ nasl_vendor_version (lex_ctxt *lexic)
   return retc;
 }
 
+/**
+ * @brief Communicate to the parent process that LSC data is ready
+ *        for use in the host kb
+ *
+ * @naslfn{update_table_driven_lsc_data}
+ *
+ * @param[in] lexic  Lexical context of the NASL interpreter.
+ *
+ * @return NULL
+ */
+tree_cell *
+nasl_update_table_driven_lsc_data (lex_ctxt *lexic)
```

Comment:
In the example you provide:
```
update_table_driven_lsc_data( pkg_list: pkg_list_notus, os_release: oskey_notus );
```

but in this function you don't gather this data nor do you handle it; meaning it does still require a

```
set_kb_item( name: "ssh/login/package_list_notus", value: value);
set_kb_item( name: "ssh/login/release_notus", value: value);
```
on the nasl side making it unnecessarily complex to use; I think it would be much easier to maintain and use later on when this specific key handling is done within this method instead of enforcing the user to know internals.

Reply:
I fixed the example. The function `update_table_driven_lsc_data()` doesn't receive any parameter.

Reply:
But my whole critique that I think it is bad design is still valid? 

So instead of two calls you need with that change calls to do the same thing. 

I still think it is bad design and that it would be better to handle notus specific restrictions within that `update_table_driven_lsc_data` method.

Reply:
So for me a call with the data:
```
update_table_driven_lsc_data( pkg_list: pkg_list_notus, os_release: oskey_notus );
```

would be much easier to remember than having to call:
```
set_kb_item( name: "ssh/login/package_list_notus", value: oskey_notus);
set_kb_item( name: "ssh/login/release_notus", value: pkg_list_notus);
update_table_driven_lsc_data( );
```

Reply:
Changes introduced with ec079f7 .
However, the changes in the plugin are not so trivial as setting the key or not, since other function are used, eg. in case of rpms, the function register_rpms is used for extra error and error return.